### PR TITLE
Gaia add retries on list bucket

### PR DESCRIPTION
**Pull request by aholdengreber**

### CODE REVIEW

**Comment by romange** [comment]
@aholdengreber   why did close?

**Comment by aholdengreber** [comment]
@romange three are some changes in this PR which made not by me..  I am not familiar with all maybe you want to merge them..

File: util/gce/gcs.cc [hunk 1]
```diff
@@ -318,10 +318,13 @@ template <typename RespBody> Status GCS::SendWithToken(Request* req, Response<Re
 
     if (IsUnauthorized(*resp)) {
       RETURN_IF_ERROR(RefreshToken(req));
-      *resp = Response<RespBody>{};
-      continue;
     }
+    *resp = Response<RespBody>{};
```
Comment (aholdengreber) [inline util/gce/gcs.cc:322]: Here is my change.. When calling List we once in a while get 503 error. I wanted to check if retries will help. before the fatal log was inside the for loop and we crash after the first try

**Review by romange — APPROVED** [review]
Thanks!

**Pull request merged**